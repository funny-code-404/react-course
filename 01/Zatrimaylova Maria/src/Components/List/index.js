// import React from 'react';
// // import './List.css'
// import ListItem from './ListItem.js';

// class List extends React.Component {

//     data = ['Menu', 'About us', 'Todays Deals', 'Customer Servise', 'Contacts']; 

//     renderList() {
//         const ListItems = this.data.map((text, index) => {
//             return <ListItem key={index}>{text}</ListItem>
//         })
//         return ListItems;
//     }

//     render() {
//         return(
//             <ul className="list">
//                 {this.renderList()}
//             </ul>
//         ) 

//     }
// }

// export default List;