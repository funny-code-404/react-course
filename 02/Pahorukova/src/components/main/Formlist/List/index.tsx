import {ReactNode} from 'react';


type Hotel = {
  id: string,
  name: string,
  city: string,
  country: string,
  imageUrl: string,
};

type Props = {
  data: Hotel[];
};

const List = ({data}: Props) => { 
  return (
    <div className="hotels-available">
       <div className="container">
         {data.length?
           <ul className="hotels__list">
            {data.map(
           (hotel: Hotel): ReactNode=> (
              <li key = {hotel.id} className="hotels__item">
                 <img className="hotel__photo" src={hotel.imageUrl} alt={hotel.name} />
                 <a href ="/" className="hotel__title">{hotel.name}</a>
                 <p className="hotel__location">{hotel.city}, {hotel.country}</p>
              </li>  
           )
          )}
          </ul> :
          <h2 className='hotels__not-found'>Unfortunately, nothing was found. Choose another hotel!</h2>
        }
       </div>
    </div>
  )
    
    };

export default List;