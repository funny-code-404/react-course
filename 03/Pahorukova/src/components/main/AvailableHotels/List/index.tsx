import {ReactNode} from 'react';
import { Hotel } from '../index';

type Props = {
  data: Hotel[];
};

const List = ({data}: Props) => { 
  return (
    <div className="hotels-available">
       <div className="container">
         {data.length?
           <ul className="hotels__list">
            {data.map(
              (hotel: Hotel): ReactNode=> (
                <li key = {hotel.id} className="hotels__item">
                   <img className="hotel__photo" src={hotel.imageUrl} alt={hotel.name} />
                    <a href ="/" className="hotel__title">{hotel.name}</a>
                    <p className="hotel__location">{hotel.city}, {hotel.country}</p>
                </li>  
             )
            )}
          </ul> :
          <div className='not-found d-flex flex-column align-center'>
            <img className='not-found__icon' src={require('../../../../imgs/search-icon.png')} alt="" />
            <p className='not-found__text'>Unfortunately, nothing was found. Choose another hotel!</p>
          </div>
        }
       </div>
    </div>
  )
    
    };

export default List;