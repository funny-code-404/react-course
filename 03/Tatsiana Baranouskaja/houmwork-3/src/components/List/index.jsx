import ListItem from "./Listitem";


const List = ({ children }) => {
    return <ul>{children}</ul>
};

List.Item = ListItem;
export default List;