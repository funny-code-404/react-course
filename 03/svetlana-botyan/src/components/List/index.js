import ListItem from './ListItem';

// eslint-disable-next-line react/prop-types
const List = ({ children }) => {
  return <ul>{children}</ul>;
};

List.Item = ListItem;

export default List;
