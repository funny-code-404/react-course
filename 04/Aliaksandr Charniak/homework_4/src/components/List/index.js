import React from 'react'
import PropTypes from 'prop-types'

import ListItem from '../ListItem'

class List extends React.Component {
  static propTypes = {
    children: PropTypes.array.isRequired,
  }

  render() {
    const { children } = this.props
    return <ul>{children}</ul>
  }
}

List.Item = ListItem

export default List
