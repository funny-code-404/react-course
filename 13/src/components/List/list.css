.list ul{
    columns: 100px 6;
    padding: 2% 0;
}