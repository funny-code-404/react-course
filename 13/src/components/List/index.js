import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import './list.css'

const List = ({data, title, path}) => {
	return  (
		<div className='list'>
			<h1>{ title }</h1>
			<ul>
				{
					Boolean(data.length) &&
						data.map(e => (
							<li key={e.id} className='li'>
								<Link to={`${path}/${e.id}`} className='link'>{e.name}</Link>
							</li>
					))
				}
			</ul>
		</div>
	)
}

List.propTypes = {
	data: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.oneOfType([
		PropTypes.number,
		PropTypes.string,
		PropTypes.arrayOf(PropTypes.oneOfType([
			PropTypes.number,
			PropTypes.string,
		])),
		PropTypes.objectOf(PropTypes.oneOfType([
			PropTypes.number,
			PropTypes.string,
			PropTypes.objectOf(PropTypes.oneOfType([
				PropTypes.number,
				PropTypes.string,
			]))
		]))
	]))),
	title: PropTypes.string.isRequired,
	path: PropTypes.string.isRequired,
}

export default List;