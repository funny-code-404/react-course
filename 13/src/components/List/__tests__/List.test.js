import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import List from "..";

configure({ adapter: new Adapter() });

describe('LIST', () => {
    const props = {
        data: [
            {
                id: 1,
                name: 'Some text...',
            },
        ], 
        title: 'Some Text title...', 
        path: '/path',
    }

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<List {...props} />);
    });

    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Component is defined', () => {
        expect(wrapper).toHaveLength(1);
    });
    it('Component should render correct title content', () => {
        expect(wrapper.find('h1').props().children).toEqual(props.title);
    });
    it('Component should render correct after props changing', () => {
        const newTitle = 'NEW';
        wrapper.setProps({ title: newTitle})
        expect(wrapper.find('h1').props().children).toEqual(newTitle);
    });
});