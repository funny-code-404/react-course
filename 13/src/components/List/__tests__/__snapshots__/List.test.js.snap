// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LIST to Match Snapshot 1`] = `
<div
  className="list"
>
  <h1>
    Some Text title...
  </h1>
  <ul>
    <li
      className="li"
      key="1"
    >
      <Link
        className="link"
        to="/path/1"
      >
        Some text...
      </Link>
    </li>
  </ul>
</div>
`;

exports[`List to Match Snapshot 1`] = `
<div
  className="list"
>
  <h1>
    Some Text title...
  </h1>
  <ul>
    <li
      className="li"
      key="1"
    >
      <Link
        className="link"
        to="/path/1"
      >
        Some text...
      </Link>
    </li>
  </ul>
</div>
`;
